lib/surface/compiler/eex_engine.ex

```diff
@@ -477,9 +506,15 @@ defmodule Surface.Compiler.EExEngine do
     props = collect_component_props(module, props)
     default_props = Surface.default_props(module)
 
+    state = %{
+      state
+      | depth: state.depth + 1,
+        context_vars: %{state.context_vars | count: state.context_vars.count + 1}
+    }
+
     [
       {add_default_bindings(component, name, let), Keyword.merge(default_props, props),
-       handle_nested_block(template, buffer, %{state | depth: state.depth + 1})}
+       handle_nested_block(template, buffer, state)}
       | handle_templates(component, tail, buffer, state)
```

Comment:
i think this will cause issues when multiple templates are provided as we're now passing the updated state to both `handle_nested_block` and the call to `handle_templates` for the next template in the list

we could either rename the updated state so it's only passed to `handle_nested_block`, or pull the update logic into a helper method and pass the result directly into `handle_nested_block`